tracing/tracing.go

```diff
@@ -6,51 +6,194 @@ package tracing
-	jaeger "github.com/uber/jaeger-client-go"
-	jaegercfg "github.com/uber/jaeger-client-go/config"
-	jaegerprom "github.com/uber/jaeger-lib/metrics/prometheus"
+	"github.com/uber/jaeger-client-go"
+	jaegerpropagator "go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/contrib/samplers/jaegerremote"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	bridge "go.opentelemetry.io/otel/bridge/opentracing"
+	jaegerotel "go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 )
 
 // ErrInvalidConfiguration is an error to notify client to provide valid trace report agent or config server
 var (
 	ErrBlankTraceConfiguration = errors.New("no trace report agent, config server, or collector endpoint specified")
 )
 
-// installJaeger registers Jaeger as the OpenTracing implementation.
-func installJaeger(serviceName string, cfg *jaegercfg.Configuration, options ...jaegercfg.Option) (io.Closer, error) {
-	metricsFactory := jaegerprom.New()
+const (
+	envSamplerParam           = "JAEGER_SAMPLER_PARAM"
+	envEndpoint               = "JAEGER_ENDPOINT"
+	envJaegerAgentHost        = "JAEGER_AGENT_HOST"
+	envJaegerAgentPort        = "JAEGER_AGENT_PORT"
+	envSamplerType            = "JAEGER_SAMPLER_TYPE"
+	envSamplerRemoteURL       = "JAEGER_SAMPLING_ENDPOINT"
+	DefaultSamplingServerPort = 5778
+)
+
+func NewFromEnv(serviceName string) (io.Closer, error) {
+	cfg, err := parseTracingConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+	}
+	if cfg.samplerRemoteURL == "" && cfg.host == "" && cfg.collectorURL == "" {
+		return nil, ErrBlankTraceConfiguration
+	}
+	return cfg.initJaegerTracerProvider(serviceName)
+}
+
+// Parse Jaeger tags from env var JAEGER_TAGS, example of TAGs format: key1=value1,key2=${value2:value3} where value2 is an env var
+// and value3 is the default value, which is optional.
+func parseAttributes(sTags string) ([]attribute.KeyValue, error) {
+	pairs := strings.Split(sTags, ",")
+	res := make([]attribute.KeyValue, 0, len(pairs))
+	for _, p := range pairs {
+		k, v, found := strings.Cut(p, "=")
+		if found {
+			k, v := strings.TrimSpace(k), strings.TrimSpace(v)
+			if strings.HasPrefix(v, "${") && strings.HasSuffix(v, "}") {
+				e, d, _ := strings.Cut(v[2:len(v)-1], ":")
+				v = os.Getenv(e)
+				if v == "" && d != "" {
+					v = d
+				}
+			}
+			res = append(res, attribute.String(k, v))
+		} else if p != "" {
+			return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+		}
+	}
+	return res, nil
+}
+
+type config struct {
+	host             string
+	collectorURL     string
+	port             int
+	samplerType      string
+	samplerRemoteURL string
+	samplerParam     float64
+	customAttributes []attribute.KeyValue
+}
+
+// This function facilitates initialization that is compatible with Jaeger's InitGlobalTracer method.
+func parseTracingConfig() (*config, error) {
```

Comment:
Rather than doing all this parsing ourselves, could we call `jaegercfg.FromEnv()` and translate the returned config to what the OTel libraries require?